Modules/Platform/MicrochipMCU-C-XC8.cmake

```diff
@@ -30,11 +30,15 @@ endif()
 
 set(CMAKE_FIND_ROOT_PATH "${MICROCHIP_XC8_PATH}")
 
-# skip compiler search and just use XC8
-find_program(CMAKE_C_COMPILER "xc8-cc"
-    PATHS "${MICROCHIP_XC8_PATH}"
-    PATH_SUFFIXES "bin"
-)
+if(NOT MICROCHIP_XC8_FORCE_NON_CC)
+    if(NOT CMAKE_C_COMPILER)
```

Comment:
`if(NOT CMAKE_C_COMPILER)` should not be required here, why did you add it?

Reply:
Hmm, well, I think this allows people to set `MICROCHIP_XC8_FORCE_NON_CC` in their CMakeLists.txt *and* override it from the outside, right?

The logic is the same as a few lines further down (already there), where it checks ` if(NOT CMAKE_C_COMPILER)`.

Happy to remove this if we feel that setting `MICROCHIP_XC8_FORCE_NON_CC` should nuke `CC`/`CXX`.

Reply:
> Hmm, well, I think this allows people to set MICROCHIP_XC8_FORCE_NON_CC in their CMakeLists.txt and override it from the outside, right?
No, this is called from toolchain file, so you can set this variable in 3 places:
 - `CMakeCache.txt`
 - Command line
 - `PreLoad.cmake`

> The logic is the same as a few lines further down (already there), where it checks `if(NOT CMAKE_C_COMPILER)`.

Yes, but here this is first `find_program` in the file, thus `CMAKE_C_COMPILER` should be not set anyway. I'm pretty sure that if it was set before CMake would not execute toolchain file in the first place.

It shouldn't cause any harm, but it wasn't in the original code and I'm 99% positive that it's not needed (though I'm hardly CMake expert...).

> Happy to remove this if we feel that setting MICROCHIP_XC8_FORCE_NON_CC should nuke `CC`/`CXX`.
I don't think that it will nuke anything because there's nothing to nuke at this point - toolchain file is supposed to find the toolchain and tell CMake how to use it. This variable should just change this module's behavior. 

If it still works for you - please remove it.

Reply:
Should be done!